numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2843,7 +2843,9 @@ static PyNumberMethods @name@_arrtype_as_number;
 static PyObject *
 bool_index(PyObject *a)
 {
-    return PyInt_FromLong(PyArrayScalar_VAL(a, Bool));
+    PyErr_SetString(PyExc_TypeError, 
```

Comment:
Instead of raising an error here, isn't it easier to remove this whole definition and not define (or explicitly set to `NULL`; not sure) `PyBoolArrType_Type.tp_as_number->nb_index`? (it is currently defined in L4114).

Reply:
Yeah, good catch